tests/acceptance/00_basics/def.json/server_dirs.cf.json

```diff
@@ -0,0 +1,7 @@
+{
+    "vars": {
+        "server_controls": {
```

Comment:
This is why I have some aversion to the deep nesting. I am looking at how this would work if we get [CFE-2084](https://tracker.mender.io/browse/CFE-2084) implemented.

This example takes two augments data sets and merges them together using `mergedata()` to merge vars together. Note how `vars.server_controls.option_a` defined in `augments_0` is missing from the merged result. Please correct me if I am wrong, but I believe this is the current behavior implemented in [PR 2515](https://github.com/cfengine/core/pull/2515).

```
    bundle agent main
    {
      vars:
        # def.json in the root of inputs
        "augments_0" data => '{ "vars": { "server_controls": { "option_a": "default", "dir_templates" : "/var/cfengine/my_templates" } } }';    
    
        # a supplementary augments file as proposed in CFE-2084
        "augments_1" data => '{ "vars": { "server_controls": { "dir_templates" : "/var/cfengine/my_templates" } } }';    
    
        "merged" data => mergedata( augments_0, augments_1 );
        "s_merged" string => format("%S", merged);
    
      reports:
    
      # Because of how mergedata works simply merging vars picked up from different
      # augments files would cause the last object to win. This is the main reason I
      # was trying to avoid deep nesting in the augments file.
    
        "CFEngine $(sys.cf_version)";
        "$(s_merged)";
    
    }
```

Here is the result:

```
    R: CFEngine 3.11.0a.334d1fc
    R: {"vars":{"server_controls":{"dir_templates":"/var/cfengine/my_templates"}}}
```